play-json/shared/src/test/scala/play/api/libs/json/MacroSpec.scala

```diff
@@ -104,61 +98,47 @@ class MacroSpec extends AnyWordSpec with Matchers with org.scalatestplus.scalach
     }
 
     "be generated for a sealed family" when {
-      implicit val simpleReads = Reads[Simple] { js =>
+      implicit val simpleReads: Reads[Simple] = Reads[Simple] { js =>
         (js \ "bar").validate[String].map(Simple(_))
       }
-      implicit val optionalReads: Reads[Optional] = Json.reads
-      implicit val familyReads: Reads[Family]     = Json.reads
+      import LoremCodec.loremReads
+      implicit val optionalReads: Reads[Optional] = Json.reads[Optional]
+      implicit val familyReads: Reads[Family]     = Json.reads[Family]
 
       val simple   = Simple("foo")
       val optional = Optional(None)
 
-      shapeless.test.illTyped("Json.reads[EmptyFamily]")
-      shapeless.test.illTyped("Json.writes[EmptyFamily]")
-      shapeless.test.illTyped("Json.format[EmptyFamily]")
+      illTyped("Json.reads[EmptyFamily]")
+      illTyped("Json.writes[EmptyFamily]")
+      illTyped("Json.format[EmptyFamily]")
 
       "using the _value syntax" in {
         val jsSimple = Json.obj(
-          "_type"  -> "play.api.libs.json.MacroSpec.Simple",
+          "_type"  -> DiscriminatorTestUtils.convert("play.api.libs.json.MacroSpec.Simple"),
           "_value" -> Json.writes[Simple].writes(simple)
         )
 
         val jsOptional = Json.obj(
-          "_type"  -> "play.api.libs.json.MacroSpec.Optional",
+          "_type"  -> DiscriminatorTestUtils.convert("play.api.libs.json.MacroSpec.Optional"),
           "_value" -> Json.writes[Optional].writes(optional)
         )
 
-        jsSimple.validate[Family].get.mustEqual(simple)
-        jsOptional.validate[Family].get.mustEqual(optional)
+        jsSimple.validate[Family].mustEqual(JsSuccess(simple))
+        jsOptional.validate[Family].mustEqual(JsSuccess(optional))
       }
 
       "using the compact syntax" in {
         val jsSimple = Json.writes[Simple].writes(simple) + (
-          "_type" -> JsString("play.api.libs.json.MacroSpec.Simple")
+          "_type" -> JsString(DiscriminatorTestUtils.convert("play.api.libs.json.MacroSpec.Simple"))
         )
 
         val jsOptional = Json.writes[Optional].writes(optional) + (
-          "_type" -> JsString("play.api.libs.json.MacroSpec.Optional")
+          "_type" -> JsString(DiscriminatorTestUtils.convert("play.api.libs.json.MacroSpec.Optional"))
         )
 
         jsSimple.validate[Family].mustEqual(JsSuccess(simple))
         jsOptional.validate[Family].mustEqual(JsSuccess(optional))
       }
-
```

Comment:
?

Reply:
Moved to `MacroScala2Spec`, because it's not supported in Scala 3.